File: macos/Sources/LinkBridgeMac/DeviceIdentity.swift
import Foundation

struct DeviceIdentity: Codable {
    let deviceId: UUID
    let deviceName: String
    let platform: String
    let protocolVersion: Int
    let features: [String]
}

final class IdentityStore {
    private let fileURL: URL

    init() {
        let appSupport = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask)[0]
        self.fileURL = appSupport
            .appendingPathComponent("LinkBridge", isDirectory: true)
            .appendingPathComponent("identity.json")
    }

    func loadOrCreate() -> DeviceIdentity {
        if let data = try? Data(contentsOf: fileURL),
           let identity = try? JSONDecoder().decode(DeviceIdentity.self, from: data) {
            return identity
        }

        let identity = DeviceIdentity(
            deviceId: UUID(),
            deviceName: Host.current().localizedName ?? "Mac",
            platform: "macos",
            protocolVersion: 1,
            features: ["file-transfer", "display-send", "display-receive"]
        )

        do {
            try FileManager.default.createDirectory(
                at: fileURL.deletingLastPathComponent(),
                withIntermediateDirectories: true
            )
            let data = try JSONEncoder().encode(identity)
            try data.write(to: fileURL, options: [.atomic])
        } catch {
            print("Warning: could not persist identity: \(error.localizedDescription)")
        }

        return identity
    }
}

File: android/app/src/main/java/com/linkbridge/android/DeviceIdentity.kt
package com.linkbridge.android

import android.content.Context
import java.util.UUID

data class DeviceIdentity(
    val deviceId: String,
    val deviceName: String,
    val platform: String = "android",
    val protocolVersion: Int = 1,
    val features: List<String> = listOf("file-transfer", "display-send", "display-receive")
)

class IdentityStore(private val context: Context) {
    private val preferences = context.getSharedPreferences("linkbridge_identity", Context.MODE_PRIVATE)

    fun loadOrCreate(): DeviceIdentity {
        val existingId = preferences.getString("device_id", null)
        val deviceId = existingId ?: UUID.randomUUID().toString().also {
            preferences.edit().putString("device_id", it).apply()
        }

        return DeviceIdentity(
            deviceId = deviceId,
            deviceName = android.os.Build.MODEL ?: "Android"
        )
    }
}

